src/main/java/org/alliancegenome/curation_api/jobs/executors/HTPExpressionDatasetAnnotationExecutor.java

```diff
@@ -51,19 +51,31 @@ public void execLoad(BulkLoadFile bulkLoadFile) {
 
 			bulkLoadFileDAO.merge(bulkLoadFile);
 			List<Long> datasetIdsLoaded = new ArrayList<>();
+			List<Long> htpAnnotationsIdsLoaded = new ArrayList<>();
 			BulkLoadFileHistory history = new BulkLoadFileHistory(htpExpressionDatasetData.getData().size());
-			createHistory(history, bulkLoadFile);
-			boolean result = runLoaddatasetid(externalDataBaseEntityService, history, dataProvider, htpExpressionDatasetData.getData(), datasetIdsLoaded, false);
+			
+			runLoad(history, dataProvider, htpExpressionDatasetData.getData(), htpAnnotationsIdsLoaded, datasetIdsLoaded);
+			
+			history.finishLoad();
+			updateHistory(history);
 		} catch (Exception e) {
 			failLoad(bulkLoadFile, e);
 			e.printStackTrace();
 		}
 	}
 
-	private boolean runLoaddatasetid(ExternalDataBaseEntityService externalDataBaseEntityService, BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDatasetData, List<Long> datasetIdsLoaded, boolean isUpdate) {
+	private void runLoad(BulkLoadFileHistory history, BackendBulkDataProvider dataProvider, List<HTPExpressionDatasetAnnotationFmsDTO> htpDataset, List<Long> htpAnnotationsIdsLoaded, List<Long> datasetIdsLoaded) {
 		ProcessDisplayHelper ph = new ProcessDisplayHelper();
 		ph.addDisplayHandler(loadProcessDisplayService);
-		ph.startProcess("External Database Entity DTO Update for " + dataProvider.name(), htpDatasetData.size());
+		ph.startProcess("HTP Expression Dataset Annotation DTO Update for " + dataProvider.name(), htpDataset.size() * 2);
+
+		runLoaddatasetid(externalDataBaseEntityService, history, dataProvider, htpDataset, datasetIdsLoaded, ph);
```

Comment:
Can these method names be changed to something like loadHtpDatasets and loadHtpDatasetAnnotations for the sake of consistency with naming of similar methods elsewhere (also not just IDs you're loading so runLoaddatasetid is a bit misleading)

Reply:
Sure Mark, will change them.